src/game/movement.rs

```diff
@@ -25,9 +32,30 @@ pub(super) fn plugin(app: &mut App) {
     );
 }
 
-#[derive(Component, Reflect, Default)]
+/// These are the movement parameters for our character controller.
+/// For now, this is only used for a single player, but it could power NPCs or other players as well.
+#[derive(Component, Reflect)]
 #[reflect(Component)]
-pub struct MovementController(pub Vec2);
+pub struct MovementController {
+    /// The direction the character wants to move in.
+    pub intent: Vec2,
+
+    /// Since Bevy's default 2D camera setup is scaled such that
+    /// one unit is one pixel, you can think of this as
+    /// "How many pixels per second should the player move?"
+    /// Note that physics engines may use different unit/pixel ratios.
```

Comment:
I feel like this is a little too far in the style of a comment that you'd find in an example, not a template where this will become the user's own code. The same can be said to a smaller degree for some of the other comments.
```suggestion
    /// Maximum speed in world units per second.
```
This is what I'd write in my own code. Maybe a balance can be struck to include a little more explanation, but still worded as though it's the user's own code.

Reply:
I strongly feel like the documentation should stay like this, as this is a question that comes up in Discord all the time. That, or people don't know what a "world unit" is and just try `5.0`, are confused why everything is extremely slow moving.

Reply:
That's fine, it's mainly the tone / perspective that I'm concerned about.
```suggestion
    /// Maximum speed in world units per second.
    /// 
    /// 1 world unit = 1 pixel with the default camera zoom.
```
for example

Reply:
[`70ce354` (#230)](https://github.com/TheBevyFlock/bevy_quickstart/pull/230/commits/70ce354bdbe1509973175176c767fd4d3248393b)